internal/venice-common/src/main/proto/VeniceReadService.proto

```diff
@@ -29,4 +54,126 @@ message VeniceServerResponse {
 
   uint32 errorCode = 6;
   string errorMessage = 7;
+}
+
+message SingleGetRequest {
+  string resourceName = 1;
+  uint32 partition = 2;
+  string key = 3;
+  bool isRetryRequest = 4;
+}
+
+message SingleGetResponse {
+  int32 statusCode = 1;
+  bytes value = 2;
+  sint32 schemaId = 3;
+  uint32 compressionStrategy = 4;
+  optional string errorMessage = 5;
+  string contentType = 6;
+  uint32 contentLength = 7;
+  uint32 rcu = 8;
+}
+
+message MultiGetRequest {
+  string resourceName = 1;
+  uint32 partition = 2;
+  bytes  keyBytes = 3;            // used for batch get
```

Comment:
Does this contain the whole slew of keys, concatenated one after the other? Should we use a collection type instead?

Reply:
Yeah, I considered that, but decided to keep it similar to the HTTP request format since changing it would also require adding / modifying the multiGet API (POST) in `TransportClient`.

However, given the benefit of avoiding deserialization if we use a collection, I’m now leaning towards making the change. Do you agree we should do it in this PR?

Reply:
I started working on this, but the changes required are quite extensive due to how tightly our code is coupled with the current abstractions. You can check out the partial changes here: [commit link](https://github.com/sushantmane/venice/commit/da641dcd248d0f1908fb0736d605cf8da2763a23).

Addressing this and the remaining three comments will require significant refactoring, which I’d prefer to handle in a separate PR, as we discussed offline.

Reply:
SG, let's leave it out of scope of the current PR.